### PR TITLE
feat: implement MainActivity with navigation setup

### LINKED ISSUE #20: Implement MainActivity with navigation setup
Set up MainActivity to manage navigation across the app's key screens and features. This includes establishing a NavHost to handle route destinations, as well as configuring specific navigation actions.

Tasks:

- Implement MainActivity with NavHost to support the main app navigation flow.
- Define start destinations and routes.

Comments:

- Implemented the top-level destinations and integrated them with the BottomNavigationMenu.
- Refactored composables to accept NavigationActions for smoother navigation handling.

### PR DESCRIPTION
**Pull request by Kenzoud**
This pull request introduces the setup for MainActivity to manage navigation across the app. It establishes a NavHost and defines the navigation flow for core screens, allowing for a structured, streamlined user experience.

Main changes:

- Configured MainActivity with a NavHost and routes for primary screens.
- Integrated NavigationActions to enable smooth transitions and proper navigation handling across screens.
- Added BottomNavigationMenu for usage in specific screens for better navigation control.

Closes Issue:

Closes #20

### CODE REVIEW

**Comment by Kenzoud** [comment]
I’m holding off on merging this PR until other pending PRs are merged into main, as they contain changes that affect shared classes. Once those are merged, I’ll pull the latest main changes into this branch, resolve any conflicts, and push the updated branch before finalizing this merge.

**Review by Ismaillat — APPROVED** [review]
Everything looks well implemented

**Pull request merged**
